service/src/main/kotlin/fi/espoo/evaka/reports/AttendanceReservationReportController.kt

```diff
@@ -204,27 +206,25 @@ private fun Database.Read.getAttendanceReservationReportByChild(
 ): List<AttendanceReservationReportByChildRow> {
     val sql =
         """
-        WITH reservations AS (
+        WITH
+        dates AS (SELECT generate_series::date AS date FROM generate_series(:start, :end, interval '1 day')),
```

Comment:
`generate_series` is probably going to cause performance problems...yes, we use it sometimes but probably shouldn't.

Here's a quick test of this report query in our production with same parameters in both cases.

Old:

```
 Nested Loop Left Join  (cost=18.72..4936.87 rows=1 width=122) (actual time=0.269..31.996 rows=82 loops=1)
...
 Planning Time: 5.996 ms
 Execution Time: 32.197 ms
```

New:

```
 Nested Loop Left Join  (cost=90368.96..100221.02 rows=1 width=146) (actual time=1249.822..1290.971 rows=264 loops=1)
...
 Planning Time: 40.757 ms
 Execution Time: 1291.270 ms
```

~1s execution time isn't terrible, but I fear it might get out of hand, especially since the date range is freely selectable and greatly affects the cost of generate_series.

I don't have any immediate good solutions, but here are two thoughts:

- add empty rows/columns in the query in Kotlin code or frontend instead. We had performance problems with one other report at some point and an important part of the fix was to fetch only the real data (using ranges) and "fill in the blanks" in frontend
- look into multiranges or do some stuff in Kotlin code, instead of "looping over" every date in the query range

Or, we can merge now and fix later if/when problems appear :man_shrugging:

Reply:
Refactored query a bit to get more readable query plan, but no performance gain unfortunately.